### DISCUSSION TITLE
Sound cards and devices - are those useful to emulate?

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#3912] [General]
OK, what will you say is the need for the following cards (# of games listed as supporting, mostly in Mobygames):

- [ ]  [Casio MT-540/CT-460/CSM-1](https://nerdlypleasures.blogspot.com/2012/04/unique-pc-hardware-game-support.html) - 14
- [ ]  [WaveJammer (PCMCIA)](https://www.mobygames.com/attributes/attribute/88/) - 81 - "Sound Blaster clone PCMCIA card for laptops"
- [ ]  [ARIA Chipset](https://www.mobygames.com/attributes/attribute/567/) (Prometheus ARIA16?, [1](http://www.faqs.org/faqs/PCsoundcards/aria/faq/), [2](https://www.vogons.org/viewtopic.php?t=65258)) - 61
- [ ]  [Reveal FX/32](https://www.mobygames.com/attributes/attribute/573/) - 23 - "can play MT-32/GM/GS music without a TSR and in real-mode DOS." [1](https://www.vogons.org/viewtopic.php?t=45313), [Reveal SoundFX WAVE 32 SC600](https://wiki.preterhuman.net/Reveal_SoundFX_WAVE_32_SC600) has on the box "Soundscape by Ensoniq"
- [ ]  [I/O Magic Tempo](https://www.mobygames.com/attributes/attribute/569/) - 22 - "OPL3 chip, compatible with Yamaha Gold Sound Standard (GSS). PCMCIA Soundcard with OPL3 and Wav capabilitys?" [1](https://www.vogons.org/viewtopic.php?t=30840&start=460), [2](https://www.vogons.org/viewtopic.php?t=66447&p=749496)
- [ ]  [Toptek Golden 16](https://www.mobygames.com/attributes/attribute/570/) - 22 - [Toptek Golden Sound Pro 16 ES](https://wiki.preterhuman.net/Toptek_Multimedia) with TMC chipset and Crystal CS4231 (WSS)?
- [ ]  https://github.com/joncampbell123/dosbox-x/issues/1365 - CMS / GameBlaster Clone
- [ ]  [Ensoniq Soundscape](https://www.mobygames.com/attributes/attribute/77/) - 268 - AudioPCI is emulated by 86Box, QEMU, Bochs
- [ ] Parallel port device
  - [ ]  [DA Converter](https://www.mobygames.com/attributes/attribute/766/) - 8
  - [ ]  [Generic LPT DAC](https://www.mobygames.com/attributes/attribute/45/) - 6
  - [ ]  [MSound](https://www.mobygames.com/attributes/attribute/46/) - 9 - more than a simple DAC, however; a DSP contained within the box could be programmed to emulate mono, stereo, or quad stereo output, up to 48KHz. - https://github.com/dosbox-staging/dosbox-staging/discussions/3902
- [ ] #3959 
- [ ] #2189 
- [ ] Roland SC-88 / JV-1080 - [15](https://www.vogons.org/viewtopic.php?p=1190364#p1190364) games
- [ ] [Yamaha XG/MU](https://github.com/joncampbell123/dosbox-x/issues/3632)

Checklist to be easy for you to tick the ones you find relevant.

**Problems**
- Moby categories are unclear. For example: what is "DA Converter" (quite generic term if you ask me), is it different (and how) from "Generic LPT DAC" - and from Disney/Covox that's already emulated?
- if a card has unique feature - is it useful?

I didn't share my opinion in advance because I wanted to hear yours' and others' who are more knowledgeable first, but here it is:

- Ensoniq Soundscape - leaning NO NEED. Quite common mode for games and also supported by various cards, including AudioPCI, which is supported by many emulators/hypervisors, but more applicable for Win9x. [X issue](https://github.com/joncampbell123/dosbox-x/issues/2726).
- Windows Sound System - leaning **NEEDED**. "some games actually [sounded the best with this](https://github.com/dosbox-staging/dosbox-staging/issues/1120#issuecomment-880239327) option, because they supported OPL3 instead of OPL2 only with this card, or supported 16-bit audio only on this card". 86Box supports it. [X issue](https://github.com/joncampbell123/dosbox-x/issues/379).
- MSound = DAC + DSP - leaning **NEEDED**. Since we know of games where Covox/Disney add unique/better mode.
- Casio MIDI module - not from Moby - leaning NO NEED. It's in the explicit list of unique hardware features. More [info and samples](https://www.vogons.org/viewtopic.php?t=49707) of how it sounds, [Space Quest 3 sounds comparison](https://www.youtube.com/watch?v=ICX6hK3Se6k). Unfortunately I assume it needs major Nuked-like effort. Reddit [emulation question](https://www.reddit.com/r/dosgaming/comments/ttj33p/casio_mt540_emulation_for_space_quest_3/?rdt=64299).

> Certain games has specific support for these midi keyboards and module. The MT-540 and the CT-460 function almost identically from a programmer perspective, even though the latter offers access to more sounds through the keyboard controls. The one difference is that the CT-460 and CSM-1 assign different tones to program change numbers compared to the MT-540, and the Apogee games identify the CT-460 in their setup programs. This would mean that the wrong sounds may be played with a MT-540.

> game devlopers used them as keyboards for composing. ... Sierra ... these synths contained a rather unique feature for a MIDI device of that time: a rich sound bank of special effects for various themes (forest, ocean, street, starwars, rainy day, evening, concert hall, wild western).. "Back in the 80's, having an electronic keyboard was a huge deal. You could entertain your guests at a party or simply fool around with the demo songs."

- PCMCIA cards - leaning NO NEED. It seems they have an unique "native mode", but I doubt it's "better" that other options supported by the game. Also people struggle turning on those native modes...
- DA Converter/Generic LPT DAC - leaning NO NEED. I assume LPT devices in Staging already cover that.
- SAAYM - not from Moby - leaning NO NEED. But there are multiple people on the linked ticket ask for [Nuked-OPM](https://github.com/nukeykt/Nuked-OPM) to be added. I assume their reasons are not game-related?
- ARIA and TMC chipsets - it seems they use unique chips, but **no clue** if they add something otherwise missing :(
- Revel - **no clue** (as for ARIA/TMC), but it can "play MT-32/GM/GS music without a TSR and in real-mode DOS." - do you find that important?

_Originally posted by @Torinde in https://github.com/dosbox-staging/dosbox-staging/issues/3901, Related: https://github.com/86Box/86Box/discussions/4759_

**Comment by johnnovak** [comment]
None of these are needed at all for their "unique" features.

- Ensoniq -- bad FM emulation. That's the only unique thing about it.
- WSS -- One or two game might use 16-bit audio or stereo FM with this. Meh. Tons of effort for almost nothing.
- Casio -- only Space Quest 3 uses it. It gives you a shitty rendition of the MT-32 original OST. Why would we _need_ to emulate a complex card to hear inferior MIDI music in a _single_ game?
- All the SB clones. Can you explain why would we _need_ to emulate any SB clone that emulates a bog standard SB 2.0 or something?

> Revel - no clue (as for ARIA/TMC), but it can "play MT-32/GM/GS music without a TSR and in real-mode DOS." - do you find that important?

Man, do you understand anything about DOS audio???

And I'm out at this point...
